pilot/pkg/xds/debug.go

```diff
@@ -132,6 +132,15 @@ func (s *DiscoveryServer) InitDebug(mux *http.ServeMux, sctl *aggregate.Controll
 		ServiceDiscovery: s.MemRegistry,
 		Controller:       s.MemRegistry.Controller,
 	})
+	s.AddDebugHandlers(mux, enableProfiling, webhook)
+}
+
+func (s *DiscoveryServer) AddDebugHandlers(mux *http.ServeMux, enableProfiling bool, webhook *inject.Webhook) {
+	// Debug handlers on HTTP ports are added for backward compatibility.
+	// They will be exposed on XDS-over-TLS in future releases.
+	if !features.EnableDebugOnHTTP {
```

Comment:
If this is disabled, debug api is not exposed at all, is this what you want?

Reply:
Yes for now. I will move readiness to a different (or 15014) later and can make this localhost only later. Rightnow, we are not able to make localhost only because of readiness issue